### PR TITLE
Switch to GitHub Actions on CI

### PR DESCRIPTION
**Pull request by jtpio**
- Switch from Travis to GitHub Actions on CI
- Checks the extensions are correctly installed

### CODE REVIEW

**Comment by jtpio** [comment]
Example run in this fork: https://github.com/jtpio/jupyter-renderers/pull/1/checks?check_run_id=2004151471

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    - name: Install the extension
+      run: |
+        jlpm && jlpm run build
+
+        jupyter labextension link packages/fasta-extension --no-build
+        jupyter labextension link packages/geojson-extension --no-build
+        jupyter labextension link packages/katex-extension --no-build
+        jupyter labextension link packages/vega2-extension --no-build
+        jupyter labextension link packages/vega3-extension --no-build
+        jupyter labextension link packages/mathjax3-extension --no-build
```
Comment (jasongrout) [inline .github/workflows/main.yml:40]: These days, `install` does `link` if it is a local directory
```suggestion
        jupyter labextension install ./packages/fasta-extension --no-build
        jupyter labextension install ./packages/geojson-extension --no-build
        jupyter labextension install ./packages/katex-extension --no-build
        jupyter labextension install ./packages/vega2-extension --no-build
        jupyter labextension install ./packages/vega3-extension --no-build
        jupyter labextension install ./packages/mathjax3-extension --no-build
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - name: Install node
+      uses: actions/setup-node@v2
+      with:
+       node-version: '14.x'
+
+    - name: Install Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+
```
Comment (jasongrout) [inline .github/workflows/main.yml:26]: Let's add Python and yarn caching
```suggestion

    - uses: actions/cache@v2
      with:
        path: ~/.cache/pip
        key: ${{ runner.os }}-pip-${{ hashFiles('**/package.json') }}
        restore-keys: |
          ${{ runner.os }}-pip-
          
    - name: Get yarn (jlpm) cache directory path
      id: yarn-cache-dir-path
      run: echo "::set-output name=dir::$(jlpm cache dir)"

    - uses: actions/cache@v2
      id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
      with:
        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
        restore-keys: |
          ${{ runner.os }}-yarn-

```

Comment (jasongrout) [reply .github/workflows/main.yml:26]: oh wait, I need to do the yarn caching after the jlab install so it has jlpm....

Comment (jasongrout) [reply .github/workflows/main.yml:26]: I'll just submit a commit, which is easier.

**Review by jasongrout — APPROVED** [review]
@jtpio, your changes look great to me. If my last commit adding caching looks good to you, feel free to merge. Thanks!

**Comment by jtpio** [comment]
Thanks Jason for the review and improvements!

Would be nice to get this in so it can be used in https://github.com/jupyterlab/jupyter-renderers/pull/243 too.

> If my last commit adding caching looks good to you, feel free to merge.

Would be happy to do so, but it looks like I don't have commit rights on this repo.

**Comment by jasongrout** [comment]
> Would be happy to do so, but it looks like I don't have commit rights on this repo.

Fixed!

**Pull request merged**